test/testOpenAPIFile.js

```diff
@@ -0,0 +1,300 @@
+var path = require("path");
+var fs = require("fs");
+
+if (!OpenAPIFile) {
+	var OpenAPIFile = require("../OpenAPIFile");
+	var OpenAPIFileType = require("../OpenAPIFileType");
+	var JsonFileType = require("ilib-loctool-json/JsonFileType");
+
+	var CustomProject = require("loctool/lib/CustomProject.js");
+	var TranslationSet =  require("loctool/lib/TranslationSet.js");
+	var ResourceString =  require("loctool/lib/ResourceString.js");
+	// var ResourcePlural =  require("loctool/lib/ResourcePlural.js");
+	// var ResourceArray =  require("loctool/lib/ResourceArray.js");
+}
+
+var p = new CustomProject({
+	name: 'foo',
+	id: 'foo',
+	sourceLocale: 'en-US'
+}, './test/testfiles', {
+	locales:['ru-RU'],
+	targetDir: '.',
+	nopseudo: true,
+	openapi: {
+		schemas: ['./test/testfiles/openapi-schema.json'],
+		mappings: {
+
+		}
+	}
+});
+
+var t = new OpenAPIFileType(p);
+
+module.exports.openapifile = {
+	constuctor: {
+		testConstructor: function(test) {
+			test.expect(1);
+
+			var oaf = new OpenAPIFile({
+				project: p,
+				jsonFileType: t.jsonFileType,
+				markdownFileType: t.markdownFileType
+			});
+
+			test.ok(oaf);
+
+			test.done();
+		},
+
+		testConstructorUseTypeToGetRequiredParam: function(test) {
+			test.expect(1);
+
+			var oaf = new OpenAPIFile({
+				project: p,
+				type: t
+			});
+
+			test.ok(oaf);
+
+			test.done();
+		},
+
+		testConstructorParams: function(test) {
+			test.expect(1);
+
+			var oaf = new OpenAPIFile({
+				project: p,
+				type: t,
+				pathName: './testfiles/json/messages.json'
+			});
+
+			test.ok(oaf);
+
+			test.done();
+		},
+
+		testConstructorErrorMissingRequiredParam: function(test) {
+			test.expect(1);
+
+			test.throws(function(test) {
+				var oaf = new OpenAPIFile({
+					project: p
+				});
+			});
+
+			test.done();
+		}
+	},
+
+	parse: {
+		testParseSimpleJsonGetBySource: function(test) {
+			test.expect(5);
+
+			var oaf = new OpenAPIFile({
+				project: p,
+				type: t
+			});
+			test.ok(oaf);
+
+			var jsonToParse = '{\n' +
+					'  "string1": "This is a test description"\n' +
+					'}';
```

Comment:
Need to have some tests of actual openapi.json snippets, including some tests of snippets that have markdown embedded in them.

Reply:
Replaced random JSON with parts of real OpenAPI json structure everywhere.